cmd/vicadmin/logs.go

```diff
@@ -378,14 +378,15 @@ func findSeekPos(f *os.File) int64 {
 		// Go back nBytes from the last newline we've seen (or beginning-of-file),
 		// and read the next nBytes
 		readPos := lastnl - int64(len(buf))
-		if pos < 0 {
+		// Fix here.  Previously was "pos < 0".
+		if readPos < 0 {
 			readPos = 0
 		}
 		bufend, err := f.ReadAt(buf, readPos)
 
 		// It's OK to get io.EOF here.  Anything else is bad.
 		if err != nil && err != io.EOF {
-			log.Errorf("Error reading from logfile: %s", err)
+			log.Errorf("Error reading from file %s: %s", f.Name(), err)
 			return 0
 		}
 
```

Comment:
Why are we referencing bufend in the loop and changing it? Why not use the single byte version of index for a single byte? If you're going to do it this via ReadAt I'd phrase it this way (pseudo-code):
```
for readPos := pos; readPos >0; {
  readPos-= len(buf)
  if readPos < 0 {
    readPos = 0
  }
  bufend, err := f.ReadAt(buf, readPos)
  if err != nil ...

  // scan buffer for newlines
  for end := bufend; end >0; {
    end = bytes.LastIndexByte(buf[:end], '\n')
    if end < 0 {
      // go idiom is to check err/exit conditions first, then return/break
      break
    }

    // logic body after validation
    nlines--
    if nlines == 0 {
      // return directly with file offset
      return readPos + end
    }
  }

  // didn't find enough newlines
  return 0
}

```